### ISSUE TITLE
[codemod] react/19/replace-default-props

### ISSUE THREAD

**Issue by quarhodron** [codemod-com/codemod#1400]
I already reported the issue but maybe not in the place where you typically look - https://github.com/reactjs/react-codemod/issues/321. There are also other people experiencing the same problem. The gist is when I am running `npx codemod react/19/replace-default-props` it only removes defaultProps and not adding default props to destructured component props. It happens to all files. 
Codemod version: 0.14.2
React/19/replace-default-props: 1.0.0

### Faulty codemod

1. codemod name, e.g., `react/19/replace-default-props`
2. [source](https://github.com/reactjs/react-codemod)

#### Original
```ts
const Button = ({ size, color }) => {
    return <button style={{ color, fontSize: size }}>Click me</button>;
}

Button.defaultProps = {
    size: '16px',
    color: 'blue'
}
```

#### Expected

```ts
const Button = ({ size = '16px', color = 'blue' }) => {
    return <button style={{ color, fontSize: size }}>Click me</button>;
}
```

#### Actual

```ts
const Button = ({ size, color }) => {
    return <button style={{ color, fontSize: size }}>Click me</button>;
}
```

### Logs:

Sorry somehow but I cannot get the logs.

I run
```
codemod react/19/replace-default-props --logs
codemod --log
```
and logs are empty

### Additional context
Add any other context about the problem here. This might include the target project, an explanation of the faulty output related to business logic, relevant links, etc.

**Comment by alexbit-codemod** [comment]
thanks so much for reporting this issue. we will look into this.

**Comment by alexbit-codemod** [comment]
cc @amirabbas-gh

**Comment by alexbit-codemod** [comment]
Thanks for reporting this issue. We’re aware of it and appreciate your feedback. Due to other priorities, it might take some time before we can address it. We’ll keep this issue open and on our radar in the meantime.

**Comment by alexbit-codemod** [comment]
actually i went ahead and published a new version of this codemod to the registry and tested on the test fixtures that @quarhodron shared and it works fine now... thanks @amirabbas-gh for the fix. closing this.

**Comment by aweibell** [comment]
This still failed for me, but I suspect it may be that I do not get the latest version from the registry.

Running this command fails:
```fish
❯ codemod react/19/replace-default-props@1.0.1 --no-cache
✖ Fetching "react/19/replace-default-props@1.0.1"...
The specified command or codemod name could not be recognized.
```

When looking around trying to find out what you published, @alexbit-codemod, I got a little confused. First I only found a version of this codemod living in 
https://github.com/codemod-com/codemod/tree/main/packages/codemods/react/19/replace-default-props
...which was last updated 5 months ago. I tried to copy the code into CodemodStudio, but imports failed – I have no experience with codemod development. The imports brought me to another repository, though, and I came across the same codemod here:
https://github.com/codemod-com/commons/blob/main/codemods/react/19/replace-default-props/
...updated three weeks ago! :)

With AI in Codemod Studio, though, I was actually able to quite fast get a codemod from scratch that did not require any manual changes and seemed to do the job! Almost a bit scary that it could be that simple!
https://go.codemod.com/LmVOsWq

**Comment by mohab-sameh** [comment]
Hi @aweibell! Thanks a lot for reporting this.

You correctly used [`--no-cache`](https://docs.codemod.com/deploying-codemods/cli#param-no-cache), so it should be pulling the latest published version on the registry, which seems to be happening. However, I suspect that your issue is caused by a regression with the codemod package version specifier.

I just tried to run the codemod using `codemod react/19/replace-default-props` and was able to fetch and run the latest codemod (as I haven't cached a previous version before). However, when I tried to run a specific version `codemod react/19/replace-default-props@1.0.1`, regardless of using `--no-cache` or not, I got the following error:

<img width="966" alt="Screenshot 2025-01-06 at 6 01 30 PM" src="https://github.com/user-attachments/assets/cd80ace7-1f93-459c-bd61-45c98e813a44" />

Seems that there is a regression in the [version specifier in Codemod CLI](https://docs.codemod.com/deploying-codemods/cli#codemod-run). @amirabbas-gh can you please check this? If that's the case, we should open a separate issue and PR to fix this CLI feature.
<img width="824" alt="image" src="https://github.com/user-attachments/assets/353e0913-526a-49f4-a46c-c84cda37074e" />

I also noticed that the latest published version on Codemod Registry shows `v1.0.1`, however [the latest version specifies `v1.0.0` in `codemodrc.json`](https://github.com/codemod-com/commons/blob/main/codemods/react/19/replace-default-props/.codemodrc.json). @alexbit-codemod I believe your latest update here was not merged to main:

> actually i went ahead and published a new version of this codemod to the registry and tested on the test fixtures that @quarhodron shared and it works fine now

We can update the [source in `codemod-com/commons`](https://github.com/codemod-com/commons/blob/main/codemods/react/19/replace-default-props/.codemodrc.json) so it's consistent with the latest version.

**Comment by alexbit-codemod** [comment]
the issues for this codemod was resolved by @amirabbas-gh in these PRs
https://github.com/codemod-com/commons/pull/4
https://github.com/codemod-com/commons/pull/1

codemods sources have moved from this repo to commons repo. so for any codemod issue we will create PRs where the source resides. and for the CLI/platform issues, we will open up PRs in this repo. thanks all.

**Comment by sumanbh** [comment]
Still seeing this issue with `react/19/replace-default-props@1.0.3` - https://github.com/sumanbh/react-19-replace-default-props-codemod-bug

**Comment by mohab-sameh** [comment]
@amirabbas-gh pushed an update. @sumanbh @aweibell, can you check if everything is working well for you now.

@aweibell the issue with the version specifier still persists and will be tracked [here](https://github.com/codemod-com/codemod/issues/1438), but please try to run the codemod with `--no-cache` enabled for now and you should be able to pull `v1.0.3` (latest) by default.

**Comment by sumanbh** [comment]
@mohab-sameh was the update published? https://codemod.com/registry/react-19-replace-default-props still shows the old version from Jan 8. it was already tagged as `v1.0.3` so maybe that's the issue.

instead, i pulled from source and can confirm the issue is fixed.

**Comment by amirabbas-gh** [comment]
Thanks to everyone who reported this issue 🙏. After reviewing, I identified the problem and have fixed it, and published it (```v1.0.4```) ✅. You can now use this codemod with ```codemod react/19/replace-default-props``` without any issues! 🎉

The related tests for this issue have also been added to the codemod files 📝, and you can check them out! 👍
